detoks/src/main/java/nl/tudelft/trustchain/detoks/Profile.kt

```diff
@@ -3,29 +3,58 @@ package nl.tudelft.trustchain.detoks
 import android.util.Log
 import nl.tudelft.trustchain.detoks.gossiper.NetworkSizeGossiper
 
-/**
- * Basic structure for a profile entry
- */
+
 class ProfileEntry(
-    var watchTime: Long = 0, // Average watch time
-    val firstSeen: Long = System.currentTimeMillis()
+    var watched:    Boolean = false,
+    val firstSeen:  Long = System.currentTimeMillis(),
+    var uploadDate: Long = 0, // This is the torrent creation date
+    var watchTime:  Long = 0, // Average watch time
+    var duration:   Long = 0, // Video duration
+    var hopCount:   Int  = 0, // Amount of other nodes visited
+    var timesSeen:  Int  = 1, // Count of times we received it
+    var likes:      Int  = 0  // TODO: Dependent on other team
 )
 
 class Profile(
     val profiles: HashMap<String, ProfileEntry> = HashMap()
 ) {
-    fun updateEntryWatchTime(key: String, time: Long, myUpdate: Boolean) {
-        if (!profiles.contains(key)) profiles[key] = ProfileEntry()
+    private fun log(string: String) {
+        Log.i(DeToksCommunity.LOGGING_TAG, string)
+    }
```

Comment:
I meant more remove the logs because I don't know if it is necessary to print, since we already have a logging overload I usually leave only the more important ones and remove all others when merging if it is very likely not relevant for others. 

If you want to leave the logs I would make it a Log.d since it is more for debugging I think? But I guess that's more of a preference. 

I would however not make this function, because having one line with Log.i(tag, str) or having log(str) makes no difference, especially since it is a private fun for this class only, seems a bit unnecessary to call a function to do that one line